src/nanotron/models/base.py

```diff
@@ -99,6 +99,18 @@ def log_modules(self, level: int = logging.DEBUG, group: Optional[ProcessGroup]
                 rank=rank,
             )
 
+    def get_named_modules(self) -> Dict[str, nn.Module]:
+        """Return the named modules that only belongs to the current pp rank."""
+        named_parameters = list(self.get_named_params_with_correct_tied())
+        names_to_modules = {}
+        for name, _ in named_parameters:
+            try:
+                names_to_modules[name] = self.get_submodule(name.rsplit(".", 1)[0])
+            except AttributeError:
+                # NOTE: this module aren't belong to the current pp rank
+                pass
+        return names_to_modules
```

Comment:
i dont think we need this function? we can use either:
1. loop over .parameters() (it should directly loop over the parameters that were alrdy built for the pp rank)
2. loop over .named_modules() which should loop over all PipelineBlocks then try accessing .pp_block to find out the ones that are built?

Otherwise if you judge necessary to keep this func, maybe rename get_named_modules to named_modules_in_pp_rank()

Reply:
i go for the 2nd approach